### ISSUE TITLE
Default export function expression regression found under very specific circumstance

### ISSUE THREAD

**Issue by dsifford** [Microsoft/TypeScript-TmLanguage#474]
**[TS and JS Grammar Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.typescript-javascript-grammar) version:**  0.0.22 / latest

Hi there,

I think I may have found a regression in the language under a very specific circumstance. The example below should trigger this issue for you.

**Issue**

```ts
export default ({
  foo = [],
//^^^source.ts meta.export.default.ts meta.objectliteral.ts
  bar = '',
  baz = true,
}: any) => baz;
```

Annotated above are the scopes that the lines are receiving.

If the same expression is written on one line, here's the scopes that it receives (which all appear to be correct)

**No Issue**
```ts
export default ({foo = [], bar = '', baz = true, }: any) => baz;
//               ^^^source.ts meta.export.default.ts meta.arrow.ts meta.parameters.ts meta.parameter.object-binding-pattern.ts variable.parameter.ts
```

Let me know if I can clarify further

Thanks again for all your amazing work on the this! 😄

**Comment by sheetalkamat** [comment]
This is by design because of limitation of tmLanguage grammar syntax. The lookups aren't multiline which means at first line we cant certainly say that its a arrow function and hence the failure to colorize.
